src/main/resources/googlepubsub-config/getAccessTokenFromRefreshToken.xml

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+-->
+<template name="getAccessTokenFromRefreshToken" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="clientId"
+               description="clientId of your App, given by google console when you registered your application"/>
+    <parameter name="clientSecret"
+               description="clientSecret of your App, given by google console when you registered your application"/>
+    <parameter name="refreshToken" description="Refresh token to exchange with an access token."/>
+    <sequence>
+        <property name="googlepubsub.refreshToken" expression="$func:refreshToken"/>
+        <property name="googlepubsub.clientId" expression="$func:clientId"/>
+        <property name="googlepubsub.clientSecret" expression="$func:clientSecret"/>
+        <payloadFactory media-type="json">
+            <format>
+                {
+                "refresh_token": "$1",
+                "grant_type": "refresh_token",
+                "client_secret": "$2",
+                "client_id": "$3"
+                }
+            </format>
+            <args>
+                <arg expression="$ctx:googlepubsub.refreshToken"/>
+                <arg expression="$ctx:googlepubsub.clientSecret"/>
+                <arg expression="$ctx:googlepubsub.clientId"/>
+            </args>
+        </payloadFactory>
+        <property name="messageType" value="application/x-www-form-urlencoded" scope="axis2"/>
+        <call>
+            <endpoint>
+                <http method="post"
+                      uri-template="https://www.googleapis.com/oauth2/v3/token"/>
+            </endpoint>
+        </call>
+        <!-- Setting the new access token to the property. -->
+        <property name="uri.var.googlepubsub.accessToken" expression="json-eval($.access_token)"/>
+        <script language="js">
+            <![CDATA[
+            importPackage(Packages.org.apache.synapse.config);
+            var newAccessToken = mc.getProperty('uri.var.googlepubsub.accessToken');
+            var newAccessRegistryPath = mc.getProperty('uri.var.googlepubsub.accessTokenRegistryPath');
+            if(newAccessToken != null && newAccessToken != ''){
```

Comment:
The best way to check whether a string is not null, not empty and has content is to do  as follows.

if (newAccessToken) {
// do somehting.
}

Reply:
done